### PR TITLE
Migrate project to (only) Go modules

### PR DESCRIPTION
**Pull request by nrcook**
Go Modules is the standard moving forward.  This PR removes `dep` from the project and fully migrates to Go modules. 

I've also updated the Readme, removing `dep` references and tried to clean up the install section a bit.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,6 @@ script:
   - GOOS=linux GOARCH=amd64 go build -ldflags="-X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildVersion=$VERSION' -X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildTime=$time' -X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildCommit=$TRAVIS_COMMIT'" -o nancy-linux.amd64-$VERSION
   - GOOS=darwin GOARCH=amd64 go build -ldflags="-X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildVersion=$VERSION' -X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildTime=$time' -X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildCommit=$TRAVIS_COMMIT'" -o nancy-darwin.amd64-$VERSION
   - GOOS=windows GOARCH=amd64 go build -ldflags="-X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildVersion=$VERSION' -X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildTime=$time' -X 'github.com/sonatype-nexus-community/nancy/buildversion.BuildCommit=$TRAVIS_COMMIT'" -o nancy-windows.amd64-$VERSION.exe
-  - ./nancy-linux.amd64-$VERSION Gopkg.lock
```
Comment (zendern) [inline .travis.yml]: I think this should still stick around as we still support dep from a nancy standpoint and this is effectively an "integration test". Maybe just move that file into a directory and point this to it.

File: README.md [hunk 2]
```diff
@@ -61,11 +58,18 @@ Each commit to master creates a new release binary, and if you'd like to skip bu
 
 ## Development
 
-`nancy` is written using Golang 1.11, so it is best you start there.
+`nancy` is built with Go 1.11 and supports Go modules. The easiest way to get started is to clone Nancy into a directory outside of the `$GOPATH`.
```
Comment (zendern) [inline README.md]: This and `Build from source` are basically identical do we really need both??

**Comment by arschles** [comment]
I'm all for this! I was asking for it in https://github.com/sonatype-nexus-community/nancy/issues/22, and I'd love to see it 😄

**Comment by zendern** [comment]
Closing this PR as https://github.com/sonatype-nexus-community/nancy/pull/67 took care of this already

**Pull request closed without merging**